spanner/client.go

```diff
@@ -456,6 +456,38 @@ func (c *Client) ReadWriteTransaction(ctx context.Context, f func(context.Contex
 	return ts, err
 }
 
+// BeginReadWriteTransaction starts a read-write transaction. Commit() or
+// Rollback() should be called when ending a transaction. If Commit() or
+// Rollback() is not called properly, then it can cause an leaked session that
+// is borrowed from the pool but is not returned.
+//
+// This method should only be used if a custom error handling is needed. For
+// normal use cases, client.ReadWriteTransaction should be used because it has
+// robust error handlings and retries.
+func (c *Client) BeginReadWriteTransaction(ctx context.Context) (*ReadWriteTransaction, error) {
```

Comment:
I think we should include an example of how a user should normally use this method. Otherwise, I worry that users might start seeing this as the 'main' transaction method and use it as they would do in a non-cloud database. In other words: We should provide an example usage where we show how the user is expected to implement a retry-loop.

Reply:
I believe most users will start from the getting started guide and we won't show this over there. Only users who check the code docs will see this method. Also, the doc comment clearly indicates that using this method needs to deal with the errors and retires. 

This method is intended to be used by library developers, e.g., developing an ORM, who expect more flexibility on error and retry handlings. 

I am happy to provide an example usage, but I just wonder where should we put this example? doc comment? code sample? integration test?

Reply:
I will put this example usage into integration tests.

Reply:
I would prefer that we have a concrete example here as well to make it 100% clear to users how they are expected to use this method. My experience is that very many users do not realize how they need to correctly handle aborted transactions. 

I would also very much prefer if we could move this function out of the `client.go` file and into a separate struct. I agree that most users will begin with the online examples and so forth. But very quickly they will also dive into the code or their own development environment, and this method will stand out very clearly: start typing anything with Transaction and this method also pops up in the autocomplete of your development environment of choice.

The same also applies to the `Commit` and `Rollback` functions of `ReadWriteTransaction`. They will show up in the development environment of users, and it will incredibly confusing to see a `Commit` function, and then having to read the comments to find out that you are not allowed to call this method inside your transaction function when using the normal `ReadWriteTransaction`.